tests/functional/adapter/test_seeding.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+import os
+from dbt.tests.util import run_dbt, relation_from_name
+
+
+seed__schema_yml = """
+version: 2
+seeds:
+  - name: test_large_seed
+    config:
+      quote_columns: false
+      column_types:
+        id: int
+        name: varchar(64)
+        value: varchar(64)
+"""
+
+# Sample SQL to test the batch size.
+test_batch_size_sql = """
+SELECT * FROM {{ ref('test_large_seed') }} LIMIT {{ get_batch_size() }}
+"""
+
+
+class TestBatchSizeIBMDB2:
+    @property
+    def models(self):
+        return "models"
+
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        current_dir = os.path.dirname(os.path.abspath(__file__))
+        seed_data_file = os.path.join(current_dir, "test_data", "test_large_seed.csv")
+        with open(seed_data_file, "r", newline="") as file:
+            test_large_seed_csv = file.read()
+        return {
+            "test_large_seed.csv": test_large_seed_csv,
+            "schema.yml": seed__schema_yml,
+        }
+
+    @pytest.fixture(scope="class")
+    def batch_size_test(self):
+        return {
+            "test_batch_size.sql": test_batch_size_sql,
+        }
+
+    @pytest.mark.parametrize("custom,BATCH_SIZE", [(True, 200), (False, 1000)])
+    def test_get_batch_size_macro(self, project, batch_size_test, custom, BATCH_SIZE):
+        results = run_dbt(["seed"])
```

Comment:
`dbt seed` does not support --vars and it will not use the custom batch sizes

Reply:
Ah sorry, that is true. But it will on the other hand respect `vars:` in a `dbt_project.yml` file. Maybe we can follow up with setting up a class that uses a defualt project and test `dbt seed` with that? Or that class is perhaps already defined?